Automattic/jetpack
# Make devicepx library an optional theme feature

Fixes #1850 #820

#### Changes proposed in this Pull Request:

* Enqueue `https://s0.wp.com/wp-content/js/devicepx-jetpack.js` only if the theme defines support for it via `add_theme_support( 'jetpack-devicepx' )`.

* Set `jetpack-devicepx` as not supported during AMP requests.

#### Testing instructions:

* Enable the Jetpack plugin. It no longer enqueues the `https://s0.wp.com/wp-content/js/devicepx-jetpack.js` script on all pages automatically.

* Add `add_theme_support( 'jetpack-devicepx' );` to theme's `functions.php` and notice that now all front-end page loads include the `https://s0.wp.com/wp-content/js/devicepx-jetpack.js` script.

* Load an AMP page and notice that the `devicepx` library is not loaded as expected.

#### Proposed changelog entry for your changes:

Disable the `devicepx` library by default and allow themes to enable it via the `add_theme_support( 'jetpack-devicepx' );` toggle.
